### DISCUSSION TITLE
Idle trigger

### DISCUSSION THREAD

**Discussion by garlett** [LBCrion/sfwbar#234] [Ideas]
Is it possible to add an idle timeout trigger, and one resume?
Then we could replace swayidle and save some memory.
Thanks

**Comment by LBCrion** [comment]
This should be fairly trivial to implement as an sfwbar module. Looking at
swayidle it handles the ext-idle-notifier events as well a some logind
events, are you looking for support for both of these?

**Comment by LBCrion** [comment]
I implemented a very basic idled trigger support. In the latest git version, if you load an `idle` module, you will get `idled` and `resumed` triggers emitted for the respective events. You can adjust the idle timeout with `SetIdleTimeout` i.e. `SetIdleTimeout "60"` for 1min timeout.

**Reply by ffs-akl** [reply]
Can I use it instead of swayidle to trigger the swaylock lock screen?

**Reply by ffs-akl** [reply]
Is it possible to implement the functionality of the following code through configuration?
```
swayidle -w \
	timeout 300 'swaylock -f -c 000000' \
	timeout 600 'wlopm --off \*' \
	resume 'wlopm --on \*' \
	before-sleep 'swaylock -f -c 000000' >/dev/null 2>&1 &

```

**Comment by LBCrion** [comment]
Currently,  you can only have one idle timeout, but it would be easy to
extend it to multiple timeouts.

**Reply by ffs-akl** [reply]
Personally, is it really worth it for the panel to achieve IDLE? maybe someone has a pressing need for a panel to implement idle.
I think the clipboard history feature is more useful than the idle feature.

**Comment by ffs-akl** [comment]
How should I replace the swayidle in the following code with idle.widget?
```
swayidle -w \
	timeout 300 'swaylock -f -c 000000' \
	timeout 600 'wlopm --off \*' \
	resume 'wlopm --on \*' \
	before-sleep 'swaylock -f -c 000000' >/dev/null 2>&1 &
```

**Comment by LBCrion** [comment]
Function("SfwBarInit") {
  IdleTimeout "timer1", "300"
  IdleTimeout "timer2", "600"
}

TriggerAction "timer1", Exec "swaylock -f -c 00000"
TriggerAction "timer2", Exec "wlopm --off \*"
TriggerAction "resumed", Exec "wlopm --on \*"

**Reply by ffs-akl** [reply]
I tried to add this code to sfwbar.config or idle.widget, after 15 minutes, nothing happened with either sfwbar.config and idle.widget .

**Comment by LBCrion** [comment]
Did you load the offshore module?

Module("idle")

**Comment by LBCrion** [comment]
Idle module.

**Comment by ffs-akl** [comment]
I'm sure I've loaded the idle module.

**Comment by LBCrion** [comment]
Can you post your config please?

**Comment by ffs-akl** [comment]
I am sorry,   I didn't have the idle module loaded,  now it work.  thank you for help!

**Comment by garlett** [comment]
$ cat ~/.config/sfwbar/sfwbar.config 
```
Module("idle")
TriggerAction "idle_timer", Exec "wlopm --off *" 
TriggerAction "idle_resume", Exec "wlopm --on *"
Function("SfwbarInit") {
	IdleTimeout "idle_timer", "420"
	Exec "wlopm --on *" # in case we had a crash with display off
}

....
#include("cal.widget") # just to be sure that the cfg is loading
....

```
log shows the module idle loading, but no trigger event

also did not worked with exec string as: "wlopm --off \*", "echo xxx > /tmp/y" and "reboot"

any idea how can I debug it ? thx

**Comment by LBCrion** [comment]
Looks like I broke the IdleTImeout function while migration to a new function api. This should be fixed now. Can you try the latest git please?

Also, exec doesn't execute in a shell by default, so for echo, you'll want to use ``Exec "sh -c 'echo xxx>/tmp/y'"`` (although it probably doesn't matter much, since it's just for testing).

**Comment by garlett** [comment]
Great, working now, thanks

**Comment by garlett** [comment]
hi, resumed not working anymore,

idle is working with sh -c 'echo ii'

could you check ?

thanks,

p.s: could you include by default a script for idle on sfwbar ?

**Comment by LBCrion** [comment]
Hi, I renamed the trigger to "idle_resume" before release. "resumed" is a
bit too generic and can cause name collisions with other modules.

By default script, do you mean an example config? I can do that, but I need
a good example for it that preferably doesn't create additional
dependencies.

**Reply by garlett** [reply]
thanks will update, 

I mean include on the sfwbar.conf
